packages/signers/src/signable-message.ts

```diff
@@ -0,0 +1,17 @@
+import { TextEncoder } from 'text-encoding-impl';
+
+import { SignatureDictionary } from './types';
+
+/** Defines a message that needs signing and its current set of signatures if any. */
+export type SignableMessage = {
+    content: Uint8Array;
+    signatures: SignatureDictionary;
+};
+
+/**
+ * Creates a signable message from a provided content.
+ * If a string is provided, it will be UTF-8 encoded.
+ */
+export function createSignableMessage(content: string | Uint8Array, signatures: SignatureDictionary = {}) {
```

Comment:
I'll just write this comment once, rather than all over the place (eg. same applies to `createSignerFromKeyPair`); I think all of this should be immutable and frozen; the wrapper and the dictionaries. We should force folks who modify this stuff to return new objects updated in the immutable style.

So I'd:

* Freeze the signature dictionary
* Freeze the wrapper
* Do that same thing everywhere
* Write tests that assert that they're frozen
* Wherever possible write tests that assert that if the signatures don't _actually_ change, the exact same dictionary is returned. Same with the message/transaction content.

Examples of all of this can be found in `@solana/transactions`.

Reply:
Thanks!
- I marked everything as readonly. Even the transaction and signer interfaces themselves (let me know if that's too much).
- I froze the `SignatureDictionaries`, the `SignedMessages` and the created/generated `KeyPairSigner`.
- I added a `expect(x).toBeFrozenObject()` test for each `Object.freeze(x)`.
- I didn't assert the last point because I currently don't have a function that changes signatures, only functions that provide new frozen `SignatureDictionaries`.